genesyscloud/knowledge/genesyscloud_knowledge_init.go

```diff
@@ -0,0 +1,13 @@
+package knowledge
+
+import (
+	registrar "terraform-provider-genesyscloud/genesyscloud/resource_register"
+)
+
+func SetRegistrar(l registrar.Registrar) {
+	l.RegisterResource("genesyscloud_knowledge_v1_document", ResourceKnowledgeDocumentV1())
```

Comment:
moved into a package ```genesyscloud_knowledge_v1_document``` ```genesyscloud_knowledge_document_variation``` to prevent cyclic dependency. the actual refactoring was done to ```genesyscloud_knowledge_document```